frame/support/src/storage/mod.rs

```diff
@@ -1532,25 +1562,66 @@ mod test {
 	}
 
 	#[test]
-	#[should_panic(expected = "Require transaction not called within with_transaction")]
-	fn require_transaction_should_panic() {
+	fn require_transaction_should_return_error() {
 		TestExternalities::default().execute_with(|| {
-			require_transaction();
+			assert!(!has_storage_layer());
 		});
 	}
 
 	#[test]
-	fn require_transaction_should_not_panic_in_with_transaction() {
+	fn require_transaction_should_not_error_in_execute_with_storage_layer() {
 		TestExternalities::default().execute_with(|| {
-			with_transaction(|| {
-				require_transaction();
-				TransactionOutcome::Commit(())
-			});
-
-			with_transaction(|| {
-				require_transaction();
-				TransactionOutcome::Rollback(())
-			});
+			assert_ok!(execute_with_storage_layer(u8::MAX, || -> DispatchResult {
```

Comment:
You can use `assert_storage_noop!` here and probably in others (eg `require_transaction_should_return_error`)  as well.

Reply:
i am not asserting a storage noop here though. I could modify storage in this call and it would still be valid. the test here is that it should return ok, and that within the function, the storage layer should exist.